satpy/resample.py

```diff
@@ -890,3 +900,28 @@ def resample_dataset(dataset, destination_area, **kwargs):
     new_data.attrs.update(area=destination_area)
 
     return new_data
+
+
+def mask_source_lonlats(source_def, mask):
```

Comment:
This function isn't supposed to be needed anymore now that pykdtree's `query` method accepts a `mask` keyword argument.

Reply:
Seemes to work without it, so great spot.